Projects/TravelCertificate/Sources/TravelCertificateNavigation.swift

```diff
@@ -0,0 +1,150 @@
+    let specification: [TravelInsuranceContractSpecification]
+}
+
+enum TravelCertificateRouterActions: Hashable {
+    case whoIsTravelling(specifiction: TravelInsuranceContractSpecification)
+    case startDate(specification: TravelInsuranceContractSpecification)
+}
+
+enum TravelCertificateRouterActionsWithoutBackButton: Hashable {
+    case processingScreen
+}
+
+public struct TravelCertificateNavigation: View {
+    @StateObject private var vm = TravelCertificateNavigationViewModel()
+    @StateObject var router = Router()
+    private let infoButtonPlacement: ToolbarItemPlacement
+    private let openCoInsured: () -> Void
+
+    public init(
+        infoButtonPlacement: ToolbarItemPlacement,
+        openCoInsured: @escaping () -> Void
+    ) {
+        self.infoButtonPlacement = infoButtonPlacement
+        self.openCoInsured = openCoInsured
+    }
+
+    public var body: some View {
+        RouterHost(router: router) {
+            showListScreen(
+                infoButtonPlacement: infoButtonPlacement
+            )
+        }
+        .environmentObject(vm)
+        .detent(
+            item: $vm.isDocumentPresented,
+            style: .large,
+            options: .constant(.withoutGrabber)
+        ) { model in
+            PDFPreview(
+                document: .init(url: model.url, title: model.title)
+            )
+        }
+        .fullScreenCover(
+            item: $vm.isStartDateScreenPresented
+        ) { specificationModel in
+            start(with: specificationModel.specification)
+                .routerDestination(for: TravelCertificateRouterActions.self) { action in
+                    switch action {
+                    case let .whoIsTravelling(specification):
+                        showWhoIsTravelingScreen(specification: specification)
+                    case let .startDate(specification):
+                        showStartDateScreen(specification: specification)
+                    }
+                }
+                .routerDestination(
+                    for: TravelCertificateRouterActionsWithoutBackButton.self,
+                    options: .hidesBackButton
+                ) { action in
+                    switch action {
+                    case .processingScreen:
+                        openProcessingScreen()
+                    }
+                }
+                .embededInNavigation()
+                .addDismissFlow()
+        }
+    }
+
+    private func showListScreen(
+        infoButtonPlacement: ToolbarItemPlacement
+    ) -> some View {
+        ListScreen(
+            infoButtonPlacement: infoButtonPlacement
+        )
+        .withDismissButton()
+        .configureTitle(L10n.TravelCertificate.cardTitle)
+    }
+
+    @ViewBuilder
+    private func start(with specifications: [TravelInsuranceContractSpecification]) -> some View {
+        if specifications.count > 1 {
+            /* TODO: CHANGE TO PUSH? */
+            showContractsList(for: specifications)
```

Comment:
@sladan-hedvig this one was push before. Make more sense to have it full screen though?